src/py/reactpy/reactpy/backend/hooks.py

```diff
@@ -1,30 +1,45 @@
-from __future__ import annotations
+from __future__ import annotations  # nocov
 
-from collections.abc import MutableMapping
-from typing import Any
+from collections.abc import MutableMapping  # nocov
+from typing import Any  # nocov
 
-from reactpy.backend.types import Connection, Location
-from reactpy.core.hooks import create_context, use_context
-from reactpy.core.types import Context
+from reactpy._warnings import warn  # nocov
+from reactpy.backend.types import Connection, Location  # nocov
+from reactpy.core.hooks import ConnectionContext, use_context  # nocov
 
-# backend implementations should establish this context at the root of an app
-ConnectionContext: Context[Connection[Any] | None] = create_context(None)
 
-
-def use_connection() -> Connection[Any]:
+def use_connection() -> Connection[Any]:  # nocov
     """Get the current :class:`~reactpy.backend.types.Connection`."""
+    warn(
+        "The module reactpy.backend.hooks has been deprecated and will be deleted in the future. ",
+        "Call use_connection in reactpy.core.hooks instead.",
+        DeprecationWarning,
+    )
+
     conn = use_context(ConnectionContext)
-    if conn is None:  # nocov
+    if conn is None:
         msg = "No backend established a connection."
         raise RuntimeError(msg)
     return conn
 
 
-def use_scope() -> MutableMapping[str, Any]:
+def use_scope() -> MutableMapping[str, Any]:  # nocov
     """Get the current :class:`~reactpy.backend.types.Connection`'s scope."""
+    warn(
+        "The module reactpy.backend.hooks has been deprecated and will be deleted in the future. ",
+        "Call use_scope in reactpy.core.hooks instead.",
+        DeprecationWarning,
+    )
+
     return use_connection().scope
 
 
-def use_location() -> Location:
+def use_location() -> Location:  # nocov
     """Get the current :class:`~reactpy.backend.types.Connection`'s location."""
+    warn(
+        "The module reactpy.backend.hooks has been deprecated and will be deleted in the future. ",
+        "Call use_location in reactpy.core.hooks instead.",
```

Comment:
Since the hook is exposed at our top-level module, we may as well say: `Call reactpy.use_location instead.`

Same deal with the other functions.